### PR TITLE
Add letsencrypt for https

### LINKED ISSUE #3: Use HTTPS
Make sure site works on HTTPS as a best practice example, need to use letencrypt.

### PR DESCRIPTION
**Pull request by rht**
Depends on:
- [x] test that it runs on the silicondzor.com server, then the staging opt param is replaced with `https://acme-v01.api.letsencrypt.org/directory`.

Closes #3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,15 +4,15 @@ silicondzor 🇦🇲
 This is the source code of [silicondzor](http://silicondzor.com)
```
Comment (rht) [inline README.md:4]: s/http/https/ later

**Comment by fxfactorial** [comment]
This is fantastic!! I have it up and running live now. Now how to follow up? Do I have to upload some generated file to lets-encrypt? Not clear which line your talking about in reference to `staging opt param`. @rht

**Comment by rht** [comment]
Saw it running too. You just have to replace the 'staging' in https://github.com/fxfactorial/silicondzor/commit/ad4f31cca78c208efd7cc599e2a92f1990ecd36a#diff-5784859bc20a9558df9dfce578b2f673R239 with https://acme-v01.api.letsencrypt.org/directory.

**Comment by fxfactorial** [comment]
@rht yes, did it Thanks! Now I see that the site is being flagged as unknown cert by browsers, is the next step to register/upload a generated file to letsencrypt?

**Comment by rht** [comment]
Mmm...Strange it should have done automatic registration once you have moved from the staging cert server.

**Comment by rht** [comment]
The `le-sni-auto` lib should have been loaded by default (https://github.com/Daplie/node-letsencrypt/blob/21bf4f98b53a5c5caeb319e31e4fed09c1aa06d4/index.js#L255). Perhaps it could be set manually with the param `sni: require('le-sni-auto').create({})`.

**Comment by fxfactorial** [comment]
@rht 

1) logs had this error for adding that `sin` parameter, 

```
throw new TypeError("expecting a function but got " + util.classString(fn));
Dec 28 19:36:07 debian-512mb-nyc3-01 node[6277]: ^
Dec 28 19:36:07 debian-512mb-nyc3-01 node[6277]: TypeError: expecting a function but got [object Undefined]
Dec 28 19:36:07 debian-512mb-nyc3-01 node[6277]: at Function.Promise.promisify (/root/silicondzor/node_modules/bluebird/js/release/promisify.js:270:15)
```

2) How can https now be used locally while developing? What should the flow be?

**Comment by rht** [comment]
Reproduced the error message. `getCertificates` isn't defined on the le-sni-auto object creation.
Relevant: https://github.com/Daplie/letsencrypt-express/issues/77

**Comment by fxfactorial** [comment]
Can you make a new issue, PR to fix this?

Thank you so much 🤗

**Comment by rht** [comment]
> How can https now be used locally while developing? What should the flow be?

Hmm.. Once the https and cert are up and running, they should be orthogonal to other dev change which doesn't modify the https code (such as replacing the challenges/store handler)?

If you're referring specific to debugging the `sni`, https://letsencrypt.org/docs/staging-environment/.

**Comment by fxfactorial** [comment]
@rht Fixed it! I deleted the generated `~/letsencrypt` directory and restarted the node process with the new staging server and it regenerated the directory and it works now correctly as trusted.

**Comment by rht** [comment]
Yay :clap: :clap: :clap: 

How did it come across your mind that it is due to the staging cert?
Let's see if I can get the upstream fixed to always check if the cert is issued by a staging or production server.

**Comment by fxfactorial** [comment]
@rht I figured that because this was a stateful change and the only seemingly stateful change outside of the silicondzor project was the generated directory. 

Yes, good to have it fixed upstream as well. I can't believe how easy this is to add HTTPS, this is why I love the node ecosystem.

**Pull request merged**